request-manager/api/api.go

```diff
@@ -76,21 +76,16 @@ func NewAPI(appCtx app.Context, rm request.Manager, jls joblog.Store, stat statu
 	api.echo.Use(middleware.Recover())
 	api.echo.Use(middleware.Logger())
 
-	// Auth hook
+	// Auth plugin: authenticate caller. This is called before every route.
```

Comment:
Do we need to keep the SetUsername hook (it's right below this)? It seems like it'll be doing the same thing as Auth.Authenticate - returning the caller. AFAIK it's used just for setting the Request.User field when creating a request - I think it'd make sense to have that set to the caller returned by Auth.

Reply:
Yeah, thought about this. Let's leave it for now so that it's an easier way to set `proto.Request.User`. Else, users will have to implement a real object that implements the `auth.Plugin` interface just to set this. For example, until we enable auth, we can leave our existing hook in place.